docs/aqua-book/language/types.md

```diff
@@ -32,18 +39,22 @@ Aqua has two types of immutable collections:
   - The bang notation, optionally followed by a number or a variable, e.g. `arr!10`, `arr!index`, `arr!` (the same as `arg!0`).
 - Options containing none or one element of a given type. The quantifier is `?`, e.g. `?string`, `?[]Error` are all valid option types. To access an element of an option, use the bang notation, e.g. `opt!` or `opt!0`. There is a literal that represents an empty option: `nil`.
 
-### Appendable Collections
+### Appendable Streams
 
 Aqua has appendable [streams](./crdt-streams.md). The quantifier is `*`, e.g. `*?string`, `*Result`, `*[]i64` are all valid stream types. To access an element of a stream, use:
-  - The square brackets notation, with any expression of integer type inside, e.g. `stream[10]`, `stream[requests.length - 1]`
+  - The square brackets notation, with any expression of integer type inside, e.g. `stream[10]`, `stream[requests.length - 1]`.
   - The bang notation, optionally followed by a number or a variable, e.g. `stream!5`, `stream!index`, `stream!` (the same as `stream!0`). 
 Note that accessing an element of a stream could trigger [join behavior](../language/flow/parallel.md#join-behavior) on the stream.
 
-Only [data types](#data-types) can be used as elements of any collection, e.g. `[]u32`, `?[]string`, `*[]?u64` are valid types, but `[]*u32`, `?*string`, `**u64` are not.
+### Appendable Maps
+
+Appendable [maps](./crdt-maps.md) are sort of streams with arrays of elements that can be accessed by `string` key. The quantifier is `%`, e.g. `%?string`, `%Result`, `%[]i64` are all valid map types. There is various methods to access elements of map:
+  - `get` and `getStream` methods that return array of elements and stream of elements by key.
+  - `keys` and `keysStream` methods that return array of keys and stream of keys.
+  - `contains` method that returns if key is exist in map or not.
+Note that accessing an element of a map could trigger [join behavior](../language/flow/parallel.md#join-behavior) on the stream.
```

Comment:
Strictly speaking, no methods of a map trigger join behavior, do they? Some of them trigger canonicalization, this is not the same thing. Can we be more clear about what methods trigger canonicalization? How about "eventually" of `...Stream` methods? With this doc it is really not obvious why are there separate `get` and `getStream`